### PR TITLE
Add Temporal random radius graph generator

### PR DESCRIPTION
**Pull request by aurorarossi**

### CODE REVIEW

File: src/GNNGraphs/GNNGraphs.jl [hunk 1]
```diff
@@ -87,7 +87,8 @@ export rand_graph,
        rand_heterograph,
        rand_bipartite_heterograph,
        knn_graph,
-       radius_graph
+       radius_graph,
+       temporal_rand_radius_graph
```
Comment (CarloLucibello) [inline src/GNNGraphs/GNNGraphs.jl:91]: maybe "rand_temporal_radius_graph" instead?

File: src/GNNGraphs/generate.jl [hunk 2]
```diff
@@ -298,3 +298,64 @@ function radius_graph(points::AbstractMatrix, r::AbstractFloat;
...
+                               number_snapshots::Int,
+                               speed::AbstractFloat,
+                               r::AbstractFloat;
+                               self_loops = false,
+                               dir = :in,
+                               kws...)
+
+Create a random temporal graph given `number_nodes` nodes and `number_snapshots` snapshots.
+First, the positions of the nodes are randomly generated in the unit square. Two nodes are connected if their distance is less than a given radius `r`.
+For each snapshot, the new positions of the points are determined by applying a vector to it. The direction of the vector is chosen uniformly and its length is chosen uniformly in [-`speed`, `speed`]. Then the connections are recomputed.
```
Comment (CarloLucibello) [inline src/GNNGraphs/generate.jl:313]:
```suggestion
Each following snapshot is obtained by applying the same construction to new positions obtained as follows.
For each snapshot, the new positions of the points are determined by applying random independent displacement vectors to the previous positions. The direction of the displacement is chosen uniformly at random and its length is chosen uniformly in `[0, speed]`. Then the connections are recomputed.
```

File: src/GNNGraphs/generate.jl [hunk 3]
```diff
@@ -298,3 +298,64 @@ function radius_graph(points::AbstractMatrix, r::AbstractFloat;
...
+                                    r::AbstractFloat;
+                                    self_loops = false,
+                                    dir = :in,
+                                    kws...)
+    points=rand(2, number_nodes)
+    tg = Vector{GNNGraph}(undef, number_snapshots)
+    for t in 1:number_snapshots
+        tg[t] = radius_graph(points, r; graph_indicator = nothing, self_loops, dir, kws...)
+        for i in 1:number_nodes
+            ρ=2*speed*rand()-speed
```
Comment (CarloLucibello) [inline src/GNNGraphs/generate.jl:354]:
```suggestion
            ρ = 2 * speed * rand() - speed
```

**Comment by CarloLucibello** [comment]
Is there a reference paper introducing this generator?

**Comment by aurorarossi** [comment]
Yes Section B paragraphs 1 and 2 of [this paper](https://arxiv.org/pdf/2101.00414.pdf). I have added it to the docstring.

**Review by CarloLucibello — APPROVED** [review]

**Pull request merged**